File: Lab 2/Tegveer Singh/Kafka-Node/consumer.js
// Load the Kafka library from JavaScript into an object
const { Kafka } = require("kafkajs")

run();
async function run() {

    try {
        // Establish a TCP connection with the broker
        const kafka = new Kafka({
            "clientId": "myapp",
            "brokers": ["localhost:9093", "localhost:9094", "localhost:9095"]
        })

        const consumer = kafka.consumer({
            "groupId": "test"
        });
        console.log("Connecting...")
        await consumer.connect()
        console.log("Connected!")

        console.log("Subscribing to topic");
        await consumer.subscribe({
            "topic": "topic",
            "fromBeginning": true
        })

        await consumer.run({
            "eachMessage": async result => {
                console.log(`Received Message: ${result.message.value} on partition: ${result.partition}`)
            }
        })
    } catch (ex) {
        console.log(`An Error occured ${ex}`)
    } finally {

    }
}

File: lab2Shayan/kafka-python/consumer.py
from kafka import KafkaConsumer

group = raw_input("Enter value for consumer group, or N \n") 

if (group.lower() == "n"): \
    consumer = KafkaConsumer('Shayan', bootstrap_servers=["localhost:9093", "localhost:9094", "localhost:9095"]) 
else:
    consumer = KafkaConsumer('Shayan', group_id=group, bootstrap_servers=["localhost:9093", "localhost:9094", "localhost:9095"]) 
for message in consumer:
  
    print ("Topic= %s Partition=%d Offset=%d Key=%s Value=%s" % (message.topic, message.partition,
                                          message.offset, message.key,
                                          message.value))

consumer.close()